project/src/main/java/com/google/ensemble/servlets/UploadServlet.java

```diff
@@ -68,6 +68,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String project = request.getParameter(INPUT_NAME_PROJECT);
 
     List<Map<String, Object>> docDataList = getFilesData(request, INPUT_NAME_FILES);
+    if (docDataList == null) {
+      System.out.println("Could not find uploaded files");
+      return;
```

Comment:
Should you maybe return an HTTP status code to indicate what went wrong? https://en.wikipedia.org/wiki/List_of_HTTP_status_codes
https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html#setStatus(int)
https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html#sendError(int,%20java.lang.String)

Reply:
I agree with Justin. You can use the status code to know when to alert the user to an error with a snackbar: https://material.angular.io/components/snack-bar/overview

Reply:
Done in a rudimentary way

Reply:
Use the constant: https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html#SC_INTERNAL_SERVER_ERROR

Reply:
Done in PR #86